raft/fsm.hh

```diff
@@ -88,7 +88,7 @@ struct leader {
     // Used to access new leader to set semaphore exception
     const raft::fsm& fsm;
     // Used to limit log size
-    std::unique_ptr<seastar::semaphore> log_limiter_semaphore;
+    lw_shared_ptr<seastar::semaphore> log_limiter_semaphore;
```

Comment:
Often the sign of bad design. The semaphore protects something. How can the semaphore have a separate lifetime than what it protects?

Reply:
Also, this super-fine-grained splitting is making review harder.